nilearn/surface/surface.py

```diff
@@ -1262,33 +1266,51 @@ def __init__(self, left=None, right=None):
                 "Either left or right (or both) must be provided."
             )
 
+        # to ensure that if 1D array are passed as input
+        # they will be saved as 1D array
+        self._squeeze_on_save = {"left": False, "right": False}
+
         parts = {}
-        if left is not None:
-            if not isinstance(left, np.ndarray):
-                left = load_surf_data(left)
-            parts["left"] = left
-        if right is not None:
-            if not isinstance(right, np.ndarray):
-                right = load_surf_data(right)
-            parts["right"] = right
+        for hemi, param in zip(["left", "right"], [left, right]):
+            if param is not None:
+                if not isinstance(param, np.ndarray):
+                    param = load_surf_data(param)
+                if param.ndim == 1:
+                    param = np.array([param])
+                    self._squeeze_on_save[hemi] = True
+                parts[hemi] = param
+        self.parts = parts
+
+        self._check_parts()
```

Comment:
The issue is that if a user set some parts manually after init and does not run _check_parts they may end up having invalid parts.

We already had this problem before, but this makes it more clear.